python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -408,9 +416,11 @@ def _verify_function_choice_behavior(
                 )
 
     def _configure_function_choice_behavior(
-        self, settings: AzureAIInferenceChatPromptExecutionSettings, kernel: Kernel
+        self, settings: AzureAIInferenceChatPromptExecutionSettings, kernel: Kernel | None
```

Comment:
I don't like this change, most likely people will not call this directly, but this combined with an assert means inconsistent behavior. Prob be better to check if `kernel` is not None inside the auto invoke specific version

Reply:
This is primarily for Mypy. The way we get the kernel from the keyword arguments means the kernel could potentially be None.